HW2/P5/physics.pyx

```diff
@@ -69,17 +74,46 @@ cdef void sub_update(FLOAT[:, ::1] XY,
     ############################################################
     # SUBPROBLEM 2: use the grid values to reduce the number of other
     # objects to check for collisions.
-    for j in range(i + 1, count):
-        XY2 = &(XY[j, 0])
-        V2 = &(V[j, 0])
-        if overlapping(XY1, XY2, R):
-            # SUBPROBLEM 4: Add locking
-            if not moving_apart(XY1, V1, XY2, V2):
-                collide(XY1, V1, XY2, V2)
-
-            # give a slight impulse to help separate them
-            for dim in range(2):
-                V2[dim] += eps * (XY2[dim] - XY1[dim])
+    grid_x_bucket = <int> (XY1[0]/grid_spacing)
+    grid_y_bucket = <int> (XY1[0]/grid_spacing) 
+    grid_size = int((1.0 / grid_spacing) + 1)
+
+    #we're checking only squares in front of us. Because we aren't updating the position of
```

Comment:
How are you ensuring this? Shouldn't you have a comparison between grids? Like an ordering, if one greater than the other. This would also ensure you dont have grid locking.